### PR TITLE
add method to set a fixed container mac address

### PR DESCRIPTION
**Pull request by seb1992**
Add method 'WithMacAddress' to set a fixed mac address to a container in TestContainersBuilder. As background information to why I need this: I want to test software that requires a license key that is tied to the mac address.

### CODE REVIEW

File: src/Testcontainers/Builders/ITestcontainersBuilder.cs [hunk 1]
```diff
@@ -69,6 +69,14 @@ public interface ITestcontainersBuilder<out TDockerContainer> : IAbstractBuilder
     [PublicAPI]
     ITestcontainersBuilder<TDockerContainer> WithHostname(string hostname);
 
+    /// <summary>
+    /// Sets the mac address of the Testcontainer.
```
Comment (HofmeisterAn) [inline src/Testcontainers/Builders/ITestcontainersBuilder.cs]:
```suggestion
    /// Sets the MAC address of the Testcontainer.
```

File: src/Testcontainers/Builders/ITestcontainersBuilder.cs [hunk 2]
```diff
@@ -69,6 +69,14 @@ public interface ITestcontainersBuilder<out TDockerContainer> : IAbstractBuilder
...
+    /// </summary>
+    /// <param name="macaddress">Testcontainers mac address.</param>
```
Comment (HofmeisterAn) [inline src/Testcontainers/Builders/ITestcontainersBuilder.cs]:
```suggestion
    /// <param name="macAddress">Testcontainers MAC address.</param>
```

File: src/Testcontainers/Builders/ITestcontainersBuilder.cs [hunk 3]
```diff
@@ -69,6 +69,14 @@ public interface ITestcontainersBuilder<out TDockerContainer> : IAbstractBuilder
...
+    /// <returns>A configured instance of <see cref="ITestcontainersBuilder{TDockerContainer}" />.</returns>
+    [PublicAPI]
+    ITestcontainersBuilder<TDockerContainer> WithMacAddress(string macaddress);
```
Comment (HofmeisterAn) [inline src/Testcontainers/Builders/ITestcontainersBuilder.cs]:
```suggestion
    ITestcontainersBuilder<TDockerContainer> WithMacAddress(string macAddress);
```

File: src/Testcontainers/Configurations/Containers/ITestcontainersConfiguration.cs [hunk 4]
```diff
@@ -49,6 +49,11 @@ public interface ITestcontainersConfiguration : IDockerResourceConfiguration
     /// </summary>
     string Hostname { get; }
 
+    /// <summary>
+    /// GEts the mac address.
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Containers/ITestcontainersConfiguration.cs]:
```suggestion
    /// Gets the MAC address.
```

File: src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs [hunk 5]
```diff
@@ -32,6 +32,7 @@ public TestcontainersConfiguration(IDockerResourceConfiguration dockerResourceCo
     /// <param name="imagePullPolicy">The image pull policy.</param>
     /// <param name="name">The name.</param>
     /// <param name="hostname">The hostname.</param>
+    /// <param name="macaddress">The mac address.</param>
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs]:
```suggestion
    /// <param name="macAddress">The MAC address.</param>
```

File: src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs [hunk 6]
```diff
@@ -55,6 +56,7 @@ public TestcontainersConfiguration(IDockerResourceConfiguration dockerResourceCo
       Func<ImagesListResponse, bool> imagePullPolicy = null,
       string name = null,
       string hostname = null,
+      string macaddress = null,
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs]:
```suggestion
      string macAddress = null,
```

File: src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs [hunk 7]
```diff
@@ -118,6 +121,8 @@ public TestcontainersConfiguration(IDockerResourceConfiguration dockerResourceCo
     /// <inheritdoc />
     public string Hostname { get; }
 
+    public string MacAddress { get; }
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs]:
```suggestion
    /// <inheritdoc />
    public string MacAddress { get; }
```

File: src/Testcontainers/Builders/TestcontainersBuilder.cs [hunk 8]
```diff
@@ -112,6 +112,12 @@ public ITestcontainersBuilder<TDockerContainer> WithHostname(string hostname)
       return this.MergeNewConfiguration(new TestcontainersConfiguration(hostname: hostname));
     }
 
+    /// <inheritdoc cref="ITestcontainersBuilder{TDockerContainer}" />
+    public ITestcontainersBuilder<TDockerContainer> WithMacAddress(string macaddress)
```
Comment (HofmeisterAn) [inline src/Testcontainers/Builders/TestcontainersBuilder.cs]:
```suggestion
    public ITestcontainersBuilder<TDockerContainer> WithMacAddress(string macAddress)
```

File: src/Testcontainers/Builders/TestcontainersBuilder.cs [hunk 9]
```diff
@@ -112,6 +112,12 @@ public ITestcontainersBuilder<TDockerContainer> WithHostname(string hostname)
...
+    {
+      return this.MergeNewConfiguration(new TestcontainersConfiguration(macaddress: macaddress));
```
Comment (HofmeisterAn) [inline src/Testcontainers/Builders/TestcontainersBuilder.cs]:
```suggestion
      return this.MergeNewConfiguration(new TestcontainersConfiguration(macAddress: macAddress));
```

File: src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs [hunk 10]
```diff
@@ -80,6 +82,7 @@ public TestcontainersConfiguration(IDockerResourceConfiguration dockerResourceCo
       this.ImagePullPolicy = imagePullPolicy;
       this.Name = name;
       this.Hostname = hostname;
+      this.MacAddress = macaddress;
```
Comment (HofmeisterAn) [inline src/Testcontainers/Configurations/Containers/TestcontainersConfiguration.cs]:
```suggestion
      this.MacAddress = macAddress;
```

**Review by HofmeisterAn — CHANGES_REQUESTED** [review]
Thanks for your pull request 🙏. I suggest to rename `macaddress` to `macAddress` (maybe it is better to push these changes all at once). Can you please add a test to `TestcontainersContainerTest`?

**Comment by seb1992** [comment]
Thanks for the quick review! I executed the suggested renaming and added a test.

Viele Grüße aus dem Schwarzwald. ;)

**Review by HofmeisterAn — APPROVED** [review]
Thanks again for the contribution.

> Viele Grüße aus dem Schwarzwald. ;)

Grüße zurück 😀

**Comment by HofmeisterAn** [comment]
f2cb532db6da73d0d79190369f202f75fb9717c7.

**Pull request closed without merging**